src/libstd/sys/windows/mod.rs

```diff
@@ -151,6 +152,27 @@ fn cvt<I: PartialEq + Zero>(i: I) -> io::Result<I> {
     }
 }
 
+fn dur2timeout(dur: Duration) -> libc::DWORD {
+    // Note that a duration is a (u64, u32) (seconds, nanoseconds) pair, and the
```

Comment:
Ah yes that definitely makes sense. We don't currently have much interoperation between low-level types and standard library types in terms of conversions like this, but I can definitely see this being a windows extension trait, for exampe.